File: Problem Set 1-3_Cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int calculate_coins(int cents);

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    } while (cents < 0);
    printf("%d", calculate_coins(cents));
}

int calculate_coins(int cents)
{
    int i = 0;
    while (cents >= 25)
    {
        i++;
        cents -= 25;
    }
    while (cents >= 10)
    {
        i++;
        cents -= 10;
    }
    while (cents >= 5)
    {
        i++;
        cents -= 5;
    }
    while (cents >= 1)
    {
        i++;
        cents -= 1;
    }
    return i;
}

File: problem-sets-6/3-sentimental-cash/cash.py
import cs50

while True:
    cents = cs50.get_float("Change owed: ")
    if cents >= 0:
        break


def main():
    n = int(cents * 100)
    print(calculate_coins(n))


def calculate_coins(cents):
    i = 0
    while cents >= 25:
        i += 1
        cents -= 25

    while cents >= 10:
        i += 1
        cents -= 10

    while cents >= 5:
        i += 1
        cents -= 5

    while cents >= 1:
        i += 1
        cents -= 1

    return i


main()
